userspace/libscap/scap.c

```diff
@@ -1392,6 +1392,13 @@ static int32_t scap_handle_eventmask(scap_t* handle, uint32_t op, uint32_t ppm_s
 		break;
 	}
 
+	if (ppm_sc >= PPM_SC_MAX)
```

Comment:
Moved the checks at scap level: there were indeed no checks here, and one could provide invalid data.